pkg/core/chain/synchronizer.go

```diff
@@ -53,15 +53,17 @@ func (s *Synchronizer) inSync(currentHeight uint64, blk block.Block) (syncState,
 	}
 
 	// otherwise notify the chain (and the consensus loop)
-	s.chain.ProcessSucceedingBlock(blk)
+	if err := s.chain.ProcessSucceedingBlock(blk); err != nil {
+		s.sequencer.add(blk)
```

Comment:
Maybe it helps to leave a comment here as to why we add the block to the sequencer in case of error, in case this is not immediately clear to the reader.

Reply:
Done.